### DISCUSSION TITLE
Preventing stderr from showing in the DOM

### DISCUSSION THREAD

**Discussion by ross-spencer** [pyscript/pyscript#2006] [Q&A] [answered]
I have been using a 2022 version of Pyscript but have recently upgraded and now see stderr is output in the DOM. Does anyone know how to control this?

![image](https://github.com/pyscript/pyscript/assets/1880412/78a3c534-8439-42e8-9e66-1d61a7211915)

The script is downloaded from a wheel and its stdout is printed to the DOM, but stderr should be going to the console only. I have been looking but can't find answers in previous issues or the docs. (But of course I could be missing it)

**Comment by WebReflection** [comment, marked as answer]
you can specify `!error` in `plugins` and you'll see it only in the console and never on the DOM ... we've been discussing this forever and the conclusion is that not every PyScript user even knows the existence of devtools / console so we decided that having errors on the DOM was the easiest way to both spot these and eventually learn about devtools.

In short:

```js
<py-config>
plugins = ["!error"]
</py-config>
```

should remove that from the DOM

**Reply by ross-spencer** [reply]
Thanks @WebReflection -- is there a place in the docs that describes this?

My first workaround was to hide it in the CSS but I'm glad to be able to change this!

**Reply by WebReflection** [reply]
there should be ... any plugin we offer prefixed with `!` is not going to happen ... right now most plugins are lazy so it's not really needed to do that but for `error` we have it up and running ASAP to indeed point instantly on the DOM to errors encountered on bootstrap and later on, whenever that's possible (there are still runtime cases we cannot catch due runtimes limitations).

I am CCing @ntoll in case we'd like to have a quick amend/update around this plugins *feature*.

**Reply by ross-spencer** [reply]
> there should be 

I agree!

Looking around based on this discussion I find a pretty indirect/unclear reference to this here: https://pyscript.github.io/docs/2024.3.1/user-guide/configuration/ in an info box. Is this the most appropriate place or should it be somewhere else? e.g. plugins? -- I could also imagine more explicit reference in the DOM section or a specific error handling part of the docs?

**Reply by WebReflection** [reply]
I think that info box is actually clean enough for [the plugins section](https://pyscript.github.io/docs/2024.3.1/user-guide/configuration/#plugins) but it's true we don't explicitly state anywhere else that errors are, by default, splashed on users' pages ... that's usually not a complain, actually a feature for most users, but I fully agree we can do better there.

Still pinging @ntoll but I hope sooner than later we'll have some dedicated time to improve, cleanup, update where needed, all our docs, and this will be surely part of that effort.

Thanks for your patience, glad at least it solved your issue. Maybe we can close this as solved?
